tachyon/math/elliptic_curves/msm/algorithms/pippenger/pippenger_adapter.h

```diff
@@ -5,6 +5,7 @@
 #include <utility>
 #include <vector>
 
+#include "tachyon/base/profiler.h"
```

Comment:
I think it's good to include perfetto header directly for build optimization. Since if we modify the code in `tachyon/base/profiler.h`, it would cause re-building every place that uses `TRACE_EVENT`, but this is not we we want.

Reply:
that won't work since we need category definition from `profiler.h`. Also since we have `profiler.cc`, I believe we won't need to change the header that often.

Reply:
Since we are on that topic, I always wondered why everything is mostly on header. Is there any reason for that?